### ISSUE TITLE
Red dots appear on products created in Marketplace shops

### ISSUE THREAD

**Issue by zenweasel** [reactioncommerce/reaction#2868]
### Expected behavior
No red dots should show up on valid products

### Actual Behavior
Red dots appear on some products

![reaction](https://user-images.githubusercontent.com/72819/30466228-f5b6e360-9a0e-11e7-90b2-f7375cbcaecb.png)

### Steps to Reproduce the Behavior
1. Login as admin
1. Create a second shop
1. Login as that second shop owner
1. Create a product
1. Return to your Main shop
1. Observe that red dots appear on marketplace products

### Versions
```
Node: 7.10.0
NPM: 5.3.0
Meteor Node: 4.8.4
Meteor NPM: 4.6.1
Reaction CLI: 0.11.0
Reaction: 1.5.0
Reaction branch: marketplace
Docker: 17.06.2-ce
```

**Comment by zenweasel** [comment]
Introduced in https://github.com/reactioncommerce/reaction/pull/2795

**Comment by jm-wanja** [comment]
The variant validation acts funny specifically for the isValid status, changing from false to true when one interacts with the product but doesn't change the failing field, eg for inventoryQuantity field when the message `theQuantity is required` the isValid status is false but then when I open the product and close it without adding the quantities or interacting with the field the isValid status changes to true. This also happens when the shop is loaded on the browser the very first time or after a reload, however simply selecting a product the isValid status changes to true. I observed this for the `inventoryQuantity`  but not for the `title`. 
Seems like productvariant validation only validates isvalid being false once for some fields and returns isValid to be true when there is some interaction with a product(without fixing what was making it invalid to begin with).

**Comment by rymorgan** [comment]
This also shows up agnostic of whether you are in edit mode or not. 

![screen shot 2017-09-19 at 2 26 19 pm](https://user-images.githubusercontent.com/20409254/30616360-d57a6546-9d46-11e7-82a7-5a4cfa79256f.png)

**Comment by jm-wanja** [comment]
The custom validation for the inventoryQuantity field in the schema was using the `ReactionProduct.checkChildVariants` method which was returning the number of child variants for a parent, using `ReactionProduct.checkInventoryVariants` returns the number of inventory variants for a parent and so I feel this is what we should use to validate the field.
 However in the comments for `ReactionProduct.checkInventoryVariants` it is stated as deprecated and is to be combined with `ReactionProduct.checkChildVariants` , not sure why this was put to be that way? More information on why `ReactionProduct.checkInventoryVariants` is to be removed will be useful. Otherwise using it resolves this issue. @zenweasel

**Comment by jm-wanja** [comment]
This issue is being resolved this PRhttps://github.com/reactioncommerce/reaction/pull/2913

**Comment by jm-wanja** [comment]
The latest fix where if the product variant is failing validation, upon double click the product opens the edit variant panel is breaking on marketplace. @zenweasel to make a conclusion on whether to close this.

**Comment by zenweasel** [comment]
Closed via #3021
